samcli/lib/build/app_builder.py

```diff
@@ -629,12 +638,31 @@ def _build_function(  # pylint: disable=R1710
             specified_build_workflow = metadata.get("BuildMethod", None) if metadata else None
 
             config = get_workflow_config(runtime, code_dir, self._base_dir, specified_workflow=specified_build_workflow)
+            code_dir = (
+                str(pathlib.Path(self._base_dir, metadata.get("ProjectRootDirectory", code_dir)).resolve())
```

Comment:
nit: so it essentially means if `metadata["ProjectRootDirectory"]` is defined and `config.language` is "provided", use `metadata["ProjectRootDirectory"]`. Otherwise, use `code_dir`.

I wonder if we can these few lines more readble.

Reply:
I can update it

Reply:
resolved